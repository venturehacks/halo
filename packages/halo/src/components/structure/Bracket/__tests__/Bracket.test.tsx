import { render } from 'enzyme';
import React from 'react';

import { Bracket } from '../index';

describe('Bracket', () => {
  test('smoke', () => {
    const component = render(
      <Bracket>
        <span>Label</span>
        <button>Click me</button>
      </Bracket>,
    );
    expect(component.text()).toContain('LabelClick me');
    expect(component.hasClass('isolateFirstChild')).toBeTruthy();
  });

  test('className', () => {
    const component = render(
      <Bracket className="mb-10">
        <span>Label</span>
        <button>Click me</button>
      </Bracket>,
    );
    expect(component.hasClass('mb-10')).toBeTruthy();
  });

  test('gap', () => {
    const component = render(
      <Bracket className="gap-8">
        <span>Label</span>
        <button>Click me</button>
      </Bracket>,
    );
    expect(component.hasClass('gap-8')).toBeTruthy();
    expect(component.hasClass('gap-4')).toBeFalsy();
  });

  test('isolate last child', () => {
    const component = render(
      <Bracket isolate="last-child">
        <span>Label</span>
        <button>Click me</button>
      </Bracket>,
    );
    expect(component.hasClass('isolateLastChild')).toBeTruthy();
  });

  describe('snapshot', () => {
    test('smoke', () => {
      const component = render(
        <Bracket>
          <span>Label</span>
          <button>Click me</button>
        </Bracket>,
      );
      expect(component).toMatchSnapshot();
    });

    test('gap', () => {
      const component = render(
        <Bracket className="gap-6">
          <span>Label</span>
          <button>Click me</button>
        </Bracket>,
      );
      expect(component).toMatchSnapshot();
    });

    test('isolate last child', () => {
      const component = render(
        <Bracket isolate="last-child">
          <span>Label</span>
          <button>Click me</button>
        </Bracket>,
      );
      expect(component).toMatchSnapshot();
    });
  });
});
