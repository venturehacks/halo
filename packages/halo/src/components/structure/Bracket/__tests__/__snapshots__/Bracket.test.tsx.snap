// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bracket snapshot gap 1`] = `
<div
  class="flex flex-col md:flex-row md:items-center component isolateFirstChild gap-6"
>
  <span>
    Label
  </span>
  <button>
    Click me
  </button>
</div>
`;

exports[`Bracket snapshot isolate last child 1`] = `
<div
  class="flex flex-col md:flex-row md:items-center component isolateLastChild gap-4"
>
  <span>
    Label
  </span>
  <button>
    Click me
  </button>
</div>
`;

exports[`Bracket snapshot smoke 1`] = `
<div
  class="flex flex-col md:flex-row md:items-center component isolateFirstChild gap-4"
>
  <span>
    Label
  </span>
  <button>
    Click me
  </button>
</div>
`;
